### DISCUSSION TITLE
How can i add haspack wallet in rainbowkit

### DISCUSSION THREAD

**Discussion by snehaa-eth** [rainbow-me/rainbowkit#1922] [Q&A]
Is there any way that i can add hashpack wallet in rainbowkit so that it can detect the browser extension.I ahve tried adding custom wallet but there i need to scan the qr code ...is there any way that if extension is downloaded then automatically get detected

**Comment by 0xmkh** [comment]
@Snehagupta1907 We have a example [here](https://www.rainbowkit.com/docs/custom-wallets#examples) for how to create a custom wallet in rainbowkit. [Here](https://github.com/rainbow-me/rainbowkit/blob/main/packages/rainbowkit/src/wallets/walletConnectors/braveWallet/braveWallet.ts) is another example for injected wallet.

If you want to we can look at adding it on our end, but not sure if hashpack wallet supports EVM chains like ethereum, polygon, optimism and etc. Will take a look.
